crates/pixi_cli/src/task.rs

```diff
@@ -309,188 +306,97 @@ pub async fn execute(args: Args) -> miette::Result<()> {
-        .into_iter()
-        .map(|(env, task_names)| {
-            let task_map = task_names
-                .into_iter()
-                .flat_map(|task_name| {
-                    env.task(&task_name, Some(env.best_platform()))
-                        .ok()
-                        .map(|task| (task_name, task))
-                })
-                .collect();
-            (env, task_map)
-        })
-        .collect();
-
     print_tasks(tasks_per_env, args.summary).into_diagnostic()?;
     Ok(())
 }
 
-async fn alias_task(mut workspace: WorkspaceMut, args: AliasArgs) -> miette::Result<()> {
-    let name = &args.alias;
-    let task: Task = args.clone().into();
-    workspace.manifest().add_task(
-        name.clone(),
-        task.clone(),
-        args.platform,
-        &FeatureName::DEFAULT,
-    )?;
-    workspace.save().await.into_diagnostic()?;
-    eprintln!(
-        "{} Added alias `{}`: {}",
-        console::style("@").blue(),
-        name.fancy_display().bold(),
-        task,
-    );
-    Ok(())
-}
-
-async fn remove_tasks(mut workspace: WorkspaceMut, args: RemoveArgs) -> miette::Result<()> {
-    let mut to_remove = Vec::new();
+async fn add_task(
+    workspace_ctx: WorkspaceContext<CliInterface>,
+    args: AddArgs,
+) -> miette::Result<()> {
     let feature = args
+        .clone()
         .feature
         .map_or_else(FeatureName::default, FeatureName::from);
-    for name in args.names.iter() {
-        if let Some(platform) = args.platform {
-            if !workspace
-                .workspace()
-                .workspace
-                .value
-                .tasks(Some(platform), &feature)?
-                .contains_key(name)
-            {
-                eprintln!(
-                    "{}Task '{}' does not exist on {}",
-                    console::style(console::Emoji("❌ ", "X")).red(),
-                    name.fancy_display().bold(),
-                    console::style(platform.as_str()).bold(),
-                );
-                continue;
-            }
-        } else if !workspace
-            .workspace()
-            .workspace
-            .value
-            .tasks(None, &feature)?
-            .contains_key(name)
-        {
-            eprintln!(
-                "{}Task `{}` does not exist for the `{}` feature",
-                console::style(console::Emoji("❌ ", "X")).red(),
-                name.fancy_display().bold(),
-                console::style(&feature).bold(),
-            );
-            continue;
-        }
-
-        // Safe to remove
-        to_remove.push((name, args.platform));
-    }
 
-    let mut removed = Vec::with_capacity(to_remove.len());
-    for (name, platform) in to_remove {
-        workspace
-            .manifest()
-            .remove_task(name.clone(), platform, &feature)?;
-        removed.push(name);
-    }
+    workspace_ctx
+        .add_task(
+            args.clone().name,
+            args.clone().into(),
```

Comment:
```suggestion
            args.name.clone(),
            args.clone().into(),
```
I assume that works as well. In general, let's try to not overdo it with cloning :)